src/form-data-wrapper.js

```diff
@@ -124,15 +124,44 @@ export class PolyfillFormDataWrapper {
 }
 
 /**
- * Wrap the native FormData implementation.
+ * Wrap the native `FormData` implementation.
  *
+ * NOTE: This differs from the standard `FormData` constructor. This constructor
+ * includes a submit button if it was used to submit the `opt_form`, where
+ * the native `FormData` constructor does not include the submit button used to
+ * submit the form.
+ * {@link https://xhr.spec.whatwg.org/#dom-formdata}
  * @implements {FormDataWrapperInterface}
  */
 class NativeFormDataWrapper {
   /** @override */
   constructor(opt_form) {
     /** @private @const {!FormData} */
     this.formData_ = new FormData(opt_form);
+
+    if (!opt_form) {
+      return;
+    }
+
+    // Include the submit button to match browser default submit behavior
+    const {activeElement} = opt_form.ownerDocument;
+    const {elements} = opt_form;
+    const {length} = elements;
+    for (let i = 0; i < length; i++) {
+      const element = elements[i];
+      if (element !== activeElement) {
+        continue;
+      }
+
+      const {
+        tagName,
+        type,
+      } = element;
+      if (type == 'submit' || tagName == 'BUTTON') {
```

Comment:
We can do this check outside the for loop, right?

Reply:
It shortcuts the search through the list of elements, since form elements might not be children of the form. We should shortcut the loop if the `activeElement` is not a submit button.